sycl/test/basic_tests/host_image_accessor_read.cpp

```diff
@@ -0,0 +1,68 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+//==---- host_image_accessor_read.cpp - SYCL host image accessor check ----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <cassert>
+#include <iostream>
+
+int foo(float *image_data) {
+
+  int result[2];
+  const auto channelOrder = cl::sycl::image_channel_order::rgba;
+  const auto channelType = cl::sycl::image_channel_type::fp32;
+
+  cl::sycl::range<3> r(3, 3, 3);
+  {
+    cl::sycl::buffer<int, 1> ResultBuf(result, cl::sycl::range<1>(2));
+    cl::sycl::queue Q;
+    cl::sycl::image<3> Image_1(image_data, channelOrder, channelType, r);
+
+    cl::sycl::range<2> pitch = Image_1.get_pitch();
+
+    auto host_image_acc =
+        Image_1.template get_access<cl::sycl::float4,
+                                    cl::sycl::access::mode::read>();
+
+    auto Sampler = cl::sycl::sampler(
+        cl::sycl::coordinate_normalization_mode::unnormalized,
+        cl::sycl::addressing_mode::none, cl::sycl::filtering_mode::nearest);
+    cl::sycl::cl_int4 Coords{0, 1, 2, 0};
+    cl::sycl::cl_float4 Ret_data = host_image_acc.read(Coords);
+    cl::sycl::cl_float4 Ret_data2 = host_image_acc.read(Coords, Sampler);
+    assert((float)Ret_data.x() == 85);
+    assert((float)Ret_data.y() == 86);
+    assert((float)Ret_data.z() == 87);
+    assert((float)Ret_data.w() == 88);
+
+    assert((float)Ret_data2.x() == 85);
+    assert((float)Ret_data2.y() == 86);
+    assert((float)Ret_data2.z() == 87);
+    assert((float)Ret_data2.w() == 88);
```

Comment:
Could you check out of boundary access is handled correctly for supported sampler types?

Reply:
added the test case for it.